Use and document correct `lazyLoading.enabled` invocation.

chriskrycho opened the pull request:
Per [the current implementation][src], the correct invocation is not a boolean but an object with `enabled: boolean` on it. This PR updates both the engine itself and the guides to match that.

[src]: https://github.com/ember-engines/ember-engines/blob/1265ad153f724c531da5e42df47c52fc9f9c71da/lib/engine-addon.js#L384:L390

chriskrycho committed:
Use and document correct `lazyLoading.enabled` invocation.

rwjblue commented on lib/guide/index.js:
```diff
@@ -1,11 +1,14 @@
 /*jshint node:true*/
-const EngineAddon = require('ember-engines/lib/engine-addon');
+const EngineAddon = require("ember-engines/lib/engine-addon");
 module.exports = EngineAddon.extend({
```
These formatting changes seem unrelated, can you revert them?

chriskrycho replied:
Whoops! Yes, sorry – thought I'd caught and reverted all of those already!

chriskrycho replied:
Done and force pushed.

villander commented:
@rwjblue can you merge this, please?

rwjblue commented on lib/guide/index.js:
```diff
@@ -7,5 +7,8 @@ module.exports = EngineAddon.extend({
   },
 
-  lazyLoading: true
+  // eslint-disable-next-line ember/avoid-leaking-state-in-ember-objects
```
```suggestion
```

rwjblue commented on lib/guide/index.js:
```diff
@@ -7,5 +7,8 @@ module.exports = EngineAddon.extend({
 
-  lazyLoading: true
+  // eslint-disable-next-line ember/avoid-leaking-state-in-ember-objects
+  lazyLoading: {
```
```suggestion
  lazyLoading: Object.freeze({
```

rwjblue commented on lib/guide/index.js:
```diff
@@ -7,5 +7,8 @@ module.exports = EngineAddon.extend({
+  // eslint-disable-next-line ember/avoid-leaking-state-in-ember-objects
+  lazyLoading: {
+    enabled: true
+  }
```
```suggestion
  })
```

rwjblue reviewed (commented):
Made some suggestions, but if we freeze we don't need to disable the linting rule inline.

rwjblue committed:
Use `Object.freeze` on `lazyLoading` config.

@rwjblue noted that the ESLint rule is smart enough to handle the case
where the object is frozen -- and it does what we need.

Co-Authored-By: chriskrycho <hello@chriskrycho.com>

chriskrycho commented:
Updated, @rwjblue – thank you!

rwjblue commented:
Awesome, thank you!
